ideascube/tests/test_tags_command.py

```diff
@@ -79,3 +79,56 @@ def test_list_should_list_tags_and_slugs(capsys):
     out, err = capsys.readouterr()
     assert tag.name in out
     assert tag.slug in out
+
+
+def test_sanitize_tags():
+    foo = Tag.objects.create(name='foo')
+    Foo = Tag.objects.create(name='Foo')
+    Bar = Tag.objects.create(name='Bar') # Create a tag with upper case first.
+    bar = Tag.objects.create(name='bar')
+    bar_ = Tag.objects.create(name='bar;')
+    Bar_ = Tag.objects.create(name='Bar;')
+    tag_to_delete = Tag.objects.create(name=':')
+    clean = Tag.objects.create(name="Other:")
+    half_clean1 = Tag.objects.create(name="Other:Foo,")
+    half_clean2 = Tag.objects.create(name="Other:foo")
+
+    doc1 = DocumentFactory(tags=[foo, bar, clean])
+    doc2 = DocumentFactory(tags=[foo, Bar, clean, half_clean2])
+    doc3 = DocumentFactory(tags=[Foo, bar])
+    doc4 = DocumentFactory(tags=[Foo, Bar_, half_clean1, half_clean2])
+    doc5 = DocumentFactory(tags=[Foo, foo, bar_, Bar])
+    doc6 = DocumentFactory(tags=[Foo, foo, Bar_, tag_to_delete])
+
+    call_command('tags', 'sanitize')
+
+    for name in ('Foo', 'Bar', 'Bar;', 'bar;', ';', 'Other:', 'Other:Foo,', 'Other:foo'):
+        assert not Tag.objects.filter(name=name)
+    for name in ('foo', 'bar', 'other', 'other:foo'):
+        assert Tag.objects.get(name=name)
+
+    for tag in (foo, Bar):
+        for doc in (doc1, doc2, doc3, doc4, doc5, doc6):
+           assert tag in doc.tags.all()
+
+    for tag in (Foo, bar, bar_, Bar_, tag_to_delete, half_clean2):
+        for doc in (doc1, doc2, doc3, doc4, doc5, doc6):
+            assert tag not in doc.tags.all()
+
+    for doc in (doc1, doc2):
+        assert clean in doc.tags.all()
+    for doc in (doc3, doc4, doc5, doc6):
+        assert clean not in doc.tags.all()
+
+    for doc in (doc2, doc4):
+        assert half_clean1 in doc.tags.all()
+    for doc in (doc1, doc3, doc5, doc6):
+        assert half_clean1 not in doc.tags.all()
+
+    # We have to reload the tag from the db as django cache the attribute in
+    # the python object.
+    assert Tag.objects.get(id=foo.id).name == 'foo'
+    assert Tag.objects.get(id=Bar.id).name == 'bar'
+    assert Tag.objects.get(id=clean.id).name == 'other'
+    assert Tag.objects.get(id=half_clean1.id).name == 'other:foo'
```

Comment:
I'd just remove these lines.

What we actually care about is that only the right tags are left, and document-tag relationships were removed/fixed.

The fact that this happened by renaming some tags and keeping their `id` seems entirely irrelevant to me, it's just an implementation detail which is highly dependent on the order in which you created the tags above and the `order_by` used in the `sanitize` command.